test/functional/tools/validation_default.xml

```diff
@@ -25,6 +25,7 @@
       <output name="out_file1">
          <assert_contents>
             <has_line line="__dq__ X echo __dq__moo" />
+            <has_n_lines n="1" />
```

Comment:
Should the has line not be :+1:

__dq__ ; echo __dq__moo

Reply:
There's a basic sanitizer that is active on all ToolParameter instances, that's mentioned in https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-sanitizer.
So to get unsanitized values you need to add a Sanitizer and allow certain values.
This happens via the InputValueWrapper when accessing the variable in Cheeath at https://github.com/galaxyproject/galaxy/blob/7830c54a60369e5faf47ee1c55f2de3604e8251e/lib/galaxy/tools/parameters/basic.py#L216